run_unittests.py

```diff
@@ -1493,6 +1493,36 @@ def test_run_installed(self):
         env['LD_LIBRARY_PATH'] = installed_libdir
         self.assertEqual(subprocess.call(installed_exe, env=env), 0)
 
+    def test_components(self):
+        testdir = os.path.join(self.common_test_dir, '139 components')
+        test_data = [
+            (['-Dhello1=disabled'], 'disabled', 'enabled', 'disabled', 'enabled', 'enabled'),
+            (['-Dhello1=disabled', '-Dhello2=disabled'],
+              'disabled', 'disabled', 'disabled', 'enabled', 'enabled'),
+            (['-Dhello1=disabled', '-Dhello2=disabled',
+              '-Dhello3=enabled'], 'disabled', 'disabled', 'enabled', 'enabled', 'enabled'),
+            (['-Dsubcomponent=disabled'], 'enabled', 'enabled', 'disabled', 'disabled', 'enabled'),
+            (['-Dsubsubcomponent=disabled'], 'enabled', 'enabled', 'disabled', 'enabled', 'disabled'),
+        ]
+
+        for (options, ehello1, ehello2, ehello3, esubhello, esubsubhello) in test_data:
+            self.setUp()
+            hello1 = os.path.join(self.builddir, 'hello1')
+            hello2 = os.path.join(self.builddir, 'hello2')
+            hello3 = os.path.join(self.builddir, 'hello3')
+            hello4 = os.path.join(self.builddir, 'hello4')
```

Comment:
__[flake8]__
* [F841] local variable 'hello4' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->